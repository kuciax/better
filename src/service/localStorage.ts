import { IToken } from "../type/Token";

const key = "better";

export const setToken = (token: string, tokenExpires: string): void => {
  localStorage.setItem(
    key,
    JSON.stringify({ value: token, expires: new Date(tokenExpires) })
  );
};

export const getToken = (): IToken | undefined => {
  const token = localStorage.getItem(key);
  if (token) {
    return JSON.parse(token) as IToken;
  }
};

export const isTokenValid = () : boolean => {
  const token = getToken();
  const nowTime = new Date().getTime();
  if (token) {
    const currentTokenTime = new Date(token.expires).getTime();
    return currentTokenTime - nowTime > 0;
  }
  return false;
};
