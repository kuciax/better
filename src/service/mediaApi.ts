import axios from "axios";
import { getMediaListSucess } from "../store/actionCreators";

interface IGetMediaList {
  token: string;
}

const url = "https://thebetter.bsgroup.eu/Media/GetMediaList";

const getHeaders = (token: string) => ({
  "Content-Type": "application/json",
  Authorization: `Bearer ${token}`,
});

const data = {
  MediaListId: 2,
  IncludeCategories: false,
  IncludeImages: true,
  IncludeMedia: false,
  PageNumber: 1,
  PageSize: 10,
};

export const getMediaList = ({ token }: IGetMediaList) => async (
  dispatch: any
) => {
  const request = await axios.post(url, { headers: getHeaders(token), data });
  return dispatch(getMediaListSucess(request.data));
};
