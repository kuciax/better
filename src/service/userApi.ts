import axios from "axios";
import { signInSucess } from "../store/actionCreators";

const url = "https://thebetter.bsgroup.eu/Authorization/SignIn";

const headers = {
  "Content-Type": "application/json",
};

export const signIn = () => async (dispatch: any) => {
  const request = await axios.post(url, { headers });
  return dispatch(signInSucess(request.data.AuthorizationToken.Token));
};
