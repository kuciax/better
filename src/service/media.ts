import { IMedia } from "../type/Media";
import { fetchMediaList, fetchMediaPlayInfo } from "./api/mediaApi";
import { getToken } from "./localStorage";

 // TODO Add types to API

export const getMediaList = async (mediaListId: number) => {
  const token = getToken();
  if (token) {
    const request = await fetchMediaList(token.value, mediaListId);

    const mapImages = (Images: any) =>
      Images.map(({ ImageTypeCode, Url }: any) => ({
        type: ImageTypeCode,
        url: Url,
      }));

    const mappedData: IMedia[] = request?.data?.Entities.map(
      ({ Title, Description, Images, Id }: any) => ({
        title: Title,
        description: Description,
        images: mapImages(Images),
        mediaId: Id,
      })
    );

    return mappedData;
  }
};

export const getMediaPlayInfo = async (mediaId: string) => {
  const token = getToken();
  if (token) {
    const request = await fetchMediaPlayInfo(token.value, mediaId);
    return request?.data?.ContentUrl;
  }
};
