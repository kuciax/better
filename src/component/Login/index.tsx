import styled from "styled-components";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import { useDispatch } from "react-redux";
import { signIn } from "../../service/userApi"

interface ILoginProps {
  className?: string;
}

const Login = ({ className }: ILoginProps) => {

  return (
    <div className={className}>
      <Typography variant="h4">Sign in</Typography>
      <form noValidate autoComplete="off" onSubmit={(e)=> { e.preventDefault(); 
      
      
   signIn();   }} >
        <div className="form">
          <TextField className="username" id="standard-basic" label="Username" />
          <TextField id="standard-basic" type="password" label="Password" />
          <Button variant="contained" color="primary" type="submit">
            Login
        </Button>
        </div>
      </form>
    </div>
  )
}

const StyledLogin = styled(Login)`
  width: 350px;
  height: 400px;
  padding: 60px;
  .form {
    display: flex;
    flex-direction: column;
  
    button {
      margin-top: 56px;
    }
  }
  
  p {
    margin-bottom: 48px;
  }

  .username{
    margin-bottom : 16px;
  }
`;

export default StyledLogin;
