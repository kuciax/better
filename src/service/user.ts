import { fetchSignIn } from "./api/userApi";
import { setToken } from "./localStorage";

// TODO Add types to API

export const signIn = async (userName: string, password: string) => {
  const request = await fetchSignIn(userName, password);
  const { Token, TokenExpires } = request.data.AuthorizationToken;
  setToken(Token, TokenExpires);
};
