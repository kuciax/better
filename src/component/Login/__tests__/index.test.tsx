import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Login from "..";
import userEvent from "@testing-library/user-event";
import { signIn } from "../../../service/user";

jest.mock("react-router", () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

jest.mock("../../../service/user", () => ({
  signIn: jest.fn(),
}));

describe("When testing Login", () => {
  // given

  const renderLogin = () => render(<Login />);

  it("should properly render and match snapshot", () => {
    // when
    const { container } = renderLogin();
    // then
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should all inputs contain proper value when they are filled", () => {
    // given
    const typedUsername = "username";
    const typedPassword = "password";

    // when
    renderLogin();
    const userNameInput = screen.getByLabelText("Username");
    const passwordInput = screen.getByLabelText("Password");

    userEvent.type(userNameInput, typedUsername);
    userEvent.type(passwordInput, typedPassword);

    // then
    expect(screen.getByDisplayValue(typedUsername)).toBeInTheDocument();
    expect(screen.getByDisplayValue(typedPassword)).toBeInTheDocument();
  });

  it("shoud password input not be filled", () => {
    // given
    const typedUsername = "username";
    const typedPassword = "password";

    // when
    renderLogin();
    const userNameInput = screen.getByLabelText("Username");

    userEvent.type(userNameInput, typedUsername);

    // then
    expect(screen.getByDisplayValue(typedUsername)).toBeInTheDocument();
    expect(screen.queryAllByDisplayValue(typedPassword)).toHaveLength(0);
  });

  it("should call signIn fn with proper username and password", () => {
    // given
    const typedUsername = "username";
    const typedPassword = "password";

    // when
    renderLogin();
    const userNameInput = screen.getByLabelText("Username");
    const passwordInput = screen.getByLabelText("Password");
    const button = screen.getByRole("button");

    userEvent.type(userNameInput, typedUsername);
    userEvent.type(passwordInput, typedPassword);
    userEvent.click(button);

    // then
    expect(signIn).toHaveBeenCalledWith(typedUsername, typedPassword);
  });
});
