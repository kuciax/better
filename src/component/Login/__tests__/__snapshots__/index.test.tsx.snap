// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When testing Login should properly render and match snapshot 1`] = `
<div
  class="sc-bdfBwQ btTsfG"
>
  <h4
    class="MuiTypography-root MuiTypography-h4"
  >
    Sign in
  </h4>
  <div
    class="form"
  >
    <div
      class="MuiFormControl-root MuiTextField-root username"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        for="standard-basic-input"
        id="standard-basic-input-label"
      >
        Username
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          id="standard-basic-input"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        for="standard-basic-password"
        id="standard-basic-password-label"
      >
        Password
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          id="standard-basic-password"
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Login
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
